### ISSUE TITLE
Pathfinder 2E: World Time Sync causes massive lag

### ISSUE THREAD

**Issue by WBHarry** [vigoren/foundryvtt-simple-calendar#374]
**Describe the bug**
A calendar with a lot of notes causes massive slowdown when using the module in the Pathfinder 2E system, if the Pathfinder 2E: World Time Sync setting is on.

**To Reproduce**
1. Fresh Pathfinder 2E world with no mods except Simple Calendar.
2. Set the Pathfinder 2E: World Time Sync setting to true.
3. Add a bunch of notes on different days in the calendar.
4. Clicking on different days, changing month, shifting from simple to detailed calendar view and other things will be increasively slowed down depending on the amount of notes. I could easily get it up to 10 seconds of freeze.

**Expected behavior**
No lagg-freezing with the Pathfinder 2E: World Time Sync setting enabled :grin:

**Foundry Information**
 - Foundry Version: 10
 - Game System: Pathfinder 2E 4.0.0-Beta3
 - Simple Calendar Version: 2.1.10

**Browser (please complete the following information):**
 - Chrome

**Additional context**
I thought I'd follow up from the discussion on the topic at Discord here.
The 2.1.10 version removed the warnings caused by having the World Time Sync setting active, but appearantly the root cause of the non-responsivness is still there. If I would have to hazard a guess, then something is looping a -lot-, since so many warnings were originally being thrown because of a singular click (sometimes 10k warnings).

I don't know the nature of the problem, and perhaps it is because of the current Pathfinder Beta version, but at least there's an issue to follow up on here now. 😄

**Comment by vigoren** [comment]
Darn I was hoping it was just the flood of warnings slowing things down (which it can).

I had a look at it and the system changed the way I access some details to keep the 2 clocks in sync. I updated how I'm accessing them and the slow down disappeared. I tested with 1500 notes and it was nice and quick after.

I have a couple other things I want to get in the next update but hope to have it out in the next day or two.
